src/Groups/matrices/MatGrp.jl

```diff
@@ -125,6 +125,13 @@ function _as_subgroup_bare(G::MatrixGroup, H::GapObj)
   return H1
 end
 
+# We may store generators on the Oscar side or on the GAP side.
+function has_gens(G::MatrixGroup)
+  isdefined(G,:gens) && return true
+  isdefined(G, :X) || return false
+  return GAP.Globals.HasGeneratorsOfGroup(GapObj(G))::Bool
+end
```

Comment:
```suggestion
function has_gens(G::MatrixGroup)
  isdefined(G, :gens) && return true
  F = codomain(_ring_iso(G))
  GAP.Globals.IsBaseRingSupportedForClassicalMatrixGroup(F, GapObj(G.descr)) || return false
  return GAP.Globals.HasGeneratorsOfGroup(GapObj(G))::Bool
end
```
what about something like this?